tests/test_deployment/test_tasks.py

```diff
@@ -0,0 +1,201 @@
+        return ssh_client
+    return _get_ssh_client
+
+
+@pytest.fixture()
+def get_sftp_client(get_ssh_client):
+    def _get_sftp_client(os_type):
+        ssh_client = get_ssh_client(os_type)
+        return SFTPClient.from_transport(ssh_client.get_transport())
+    return _get_sftp_client
+
+
+@pytest.fixture()
+def get_host_name_by_os_type():
+    def _get_host_name_by_os_type(os_type):
+        return {
+            'windows': 'gdc-qa-io-350.lab.gdc.il.infinidat.com',
+            'linux': 'gdc-qa-io-017.lab.gdc.il.infinidat.com'
+            }[os_type]
+    return _get_host_name_by_os_type
+
+
+@pytest.fixture(scope="module")
+def remote_combadge_name():
+    random_string = str(uuid4())[:10]
+    return f"combadge_{random_string}"
+
+
+@pytest.fixture(scope="module")
+def get_remote_combadge_path(remote_combadge_name):
+    def _get_remote_combadge_path(os_type):
+        if os_type == 'windows':
+            remote_path = str(PureWindowsPath(os.path.join('C:', 'Users', 'root', 'AppData', 'Local', 'Temp', remote_combadge_name)))
+        elif os_type in ['linux', 'darwin']:
+            remote_path = str(os.path.join('/tmp', remote_combadge_name))
+        return remote_path
+    return _get_remote_combadge_path
+
+
+@pytest.fixture(scope="module")
+def get_local_combadge_path():
+    def _get_local_combadge_path(os_type):
+        [local_dir, local_name] = {
+            'windows': ['combadge_windows', 'combadge.exe'],
+            'linux': ['combadge_linux', 'combadge'],
+            'darwin': ['combadge_darwin', 'combadge'],
+        }[os_type]
+        local_path = os.path.realpath(os.path.join('combadge_assets', local_dir, local_name))
+        return local_path
+    return _get_local_combadge_path
+
+
+@pytest.fixture()
+def get_remote_file_to_beam(get_host_name_by_os_type):
+    def _get_remote_file_to_beam(os_type):
+        host_name = get_host_name_by_os_type(os_type)
+        file_path = {
+            'windows': str(PureWindowsPath(os.path.join('C:', 'Users', 'root', 'AppData', 'Local', 'Temp', 'scotty_test.txt'))),
+            'linux': 'AutoInstall',
+        }[os_type]
+        return [host_name, file_path]
+    return _get_remote_file_to_beam
+
+def get_pkey():
+    with open(os.path.expanduser('~/.ssh/infradev-id_rsa'), 'r') as f:
+        return f.read()
+
+
+@pytest.mark.parametrize("os_type", OS_TYPES)
+def test_get_os_type(os_type, get_ssh_client):
+    assert os_type == _get_os_type(get_ssh_client(os_type))
+
+
+@pytest.mark.parametrize("os_type", OS_TYPES)
+def test_get_connected_ssh_client(os_type, get_ssh_client, get_host_name_by_os_type):
+    ssh_client = get_ssh_client(os_type)
+    _, stdout, stderr = ssh_client.exec_command("hostname")
+    host_name = get_host_name_by_os_type(os_type)
+    assert stdout.read().decode().strip() == host_name
+
+
+@pytest.mark.parametrize("os_type", OS_TYPES)
+def test_upload_new_combadge(os_type, get_ssh_client):
```

Comment:
Rename to v2 (next func to v1).
Try to see if it's possible to unite the min a single test, with parametrization.